### PR TITLE
Add footer toggle to Onboarding's user-follow slide

### PR DESCRIPTION
**Pull request by maestromac**
## What type of PR is this? (check all applicable)
- [x] Feature

## Description
Created a modal footer let user know the numbers of people they are following. Also created a toggle all feature.
## Related Tickets & Documents
Resolves https://github.com/thepracticaldev/dev.to/issues/6549
## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
Kapture 2020-04-13 at 18 03 53

## Added tests?
- [x] yes

## Added to documentation?
- [x] no documentation needed

## [optional] Are there any post deployment tasks we need to perform?
n/a

### CODE REVIEW

**Comment by lisasy** [comment]
This looks great!

File: app/assets/stylesheets/preact/onboarding-modal.scss [hunk 1]
```diff
@@ -147,12 +151,29 @@
...
+  justify-content: center;
 
   .selection-status-content {
     display: block;
-    padding: $su-4;
     text-align: center;
+
+    p {
+      font-size: $su-4;
+      font-weight: $fw-bold;
```
Comment (lisasy) [inline app/assets/stylesheets/preact/onboarding-modal.scss:166]: awesome use of Crayons :)

File: app/assets/stylesheets/preact/onboarding-modal.scss [hunk 2]
```diff
@@ -147,12 +151,29 @@
...
+      line-height: 150%;
```
Comment (lisasy) [inline app/assets/stylesheets/preact/onboarding-modal.scss:167]: Since you're using Crayons variables for the other properties, you could also use one of the Crayons variables for line-height. To render a 150% line-height, you can say: `line-height: $lh-base;`

File: app/javascript/onboarding/__tests__/Onboarding.test.jsx [hunk 3]
```diff
@@ -299,12 +299,33 @@ describe('<Onboarding />', () => {
       const followUsers = onboardingSlides.find(<FollowUsers />);
 
       onboardingSlides.find('.user').first().simulate('click');
+      expect(onboardingSlides.find('p').last().text()).toBe(
+        "You're following 1 person",
+      );
+      onboardingSlides.find('.user').last().simulate('click');
+      expect(onboardingSlides.find('p').last().text()).toBe(
+        "You're following 2 people",
```
Comment (lisasy) [inline app/javascript/onboarding/__tests__/Onboarding.test.jsx:307]: nice pluralization :)

**Comment by maestromac** [comment]
@rhymes sorry I forgot to point that out. Yes, it's intentional.

File: app/assets/stylesheets/preact/onboarding-modal.scss (hunk 2 above)
Comment (maestromac) [reply app/assets/stylesheets/preact/onboarding-modal.scss:167]: ah I missed this one. Thanks!

File: app/javascript/onboarding/components/FollowUsers.jsx [hunk 4]
```diff
@@ -95,17 +95,27 @@ class FollowUsers extends Component {
...
     const { users, selectedUsers } = this.state;
     const { prev } = this.props;
+    let followingStatus;
+    if (selectedUsers < 2) {
+      followingStatus = "You're not following anyone";
+    } else {
+      const everyone =
+        selectedUsers.length === users.length ? ' (everyone)' : '';
+      const object = selectedUsers.length === 1 ? 'person' : 'people';
+      followingStatus = `You're following ${selectedUsers.length} ${object}${everyone}`;
```
Comment (vaidehijoshi) [inline app/javascript/onboarding/components/FollowUsers.jsx:105]: I find this interpolation hard to follow 😕 What if we just split this out into an `else if` for following a certain amount of people, and then one more `else if` for following everyone?

File: app/javascript/onboarding/components/FollowUsers.jsx [hunk 5]
```diff
@@ -95,17 +95,27 @@ class FollowUsers extends Component {
...
+    }
+
     return (
       <div className="onboarding-main">
         <Navigation prev={prev} next={this.handleComplete} />
-        <div className="onboarding-content">
+        <div className="onboarding-content no-padding-bottom">
```
Comment (vaidehijoshi) [inline app/javascript/onboarding/components/FollowUsers.jsx:111]:
```suggestion
        <div className="onboarding-content footer">
```
Instead of naming the class based on the behavior/logic it contains, I think it makes more sense to name it based on the actual element in that needs that style. IMO, I don't need to know that we have `no-padding-bottom` added as a class on this `div` when I look at the JSX being rendered; it could just be abstracted out into a `footer` class and when I need to know what styles the footer has, I could just figure that out from the class.

Btw, `footer` is just a suggestion, it might not be the right name 🙃

File: app/javascript/onboarding/components/FollowUsers.jsx [hunk 6]
```diff
@@ -136,10 +146,11 @@ class FollowUsers extends Component {
         </div>
         <div className="onboarding-selection-status">
           <div className="selection-status-content">
+            <p>{followingStatus}</p>
             <button type="button" onClick={() => this.handleSelectAll()}>
-              Select All 
-              {' '}
-              {selectedUsers.length === users.length ? '✅' : ''}
+              {selectedUsers.length === users.length
+                ? 'Deselect all'
```
Comment (vaidehijoshi) [inline app/javascript/onboarding/components/FollowUsers.jsx:152]: This logic is missing a specific situation where there are no users to begin with. Here's what displays in that case:
Screen Shot 2020-04-14 at 8 56 27 AM

I think we should account for this because, as #7106 suggests, some new communities won't have users to follow by default.

**Comment by vaidehijoshi** [comment]
Just a few minor things that I caught! I think aside from those, this is looking great and I can't wait to see it shipped 🚀

File: app/javascript/onboarding/components/FollowUsers.jsx (hunk 5 above)
Comment (maestromac) [reply app/javascript/onboarding/components/FollowUsers.jsx]: The habit of a Ruby developer 😆 . I updated it to `toggle-bottom`.

File: app/javascript/onboarding/components/FollowUsers.jsx (hunk 4 above)
Comment (maestromac) [reply app/javascript/onboarding/components/FollowUsers.jsx]: Great suggestion. Updated!

File: app/javascript/onboarding/components/FollowUsers.jsx (hunk 6 above)
Comment (maestromac) [reply app/javascript/onboarding/components/FollowUsers.jsx]: I updated it so that it will render no toggle button if there are no users to follow

**Comment by maestromac** [comment]
@vaidehijoshi it will now look like this when there's no users to follow 
image

File: app/javascript/onboarding/components/FollowUsers.jsx [hunk 7]
```diff
@@ -92,19 +92,41 @@ class FollowUsers extends Component {
...
+    } else if (selectedUsers.length === 1) {
+      followingStatus = "You're following 1 person";
+    } else if (selectedUsers.length === users.length) {
+      followingStatus = `You're following ${selectedUsers.length} people (everyone)`;
     } else {
-      const everyone =
-        selectedUsers.length === users.length ? ' (everyone)' : '';
-      const object = selectedUsers.length === 1 ? 'person' : 'people';
-      followingStatus = `You're following ${selectedUsers.length} ${object}${everyone}`;
+      followingStatus = `You're following ${selectedUsers.length} people`;
```
Comment (vaidehijoshi) [inline app/javascript/onboarding/components/FollowUsers.jsx:105]: I hope we can one day use i18n and then never have to do these pluralizations again! 🙃

File: app/javascript/onboarding/components/FollowUsers.jsx [hunk 8]
```diff
@@ -92,19 +92,41 @@ class FollowUsers extends Component {
...
+    }
+
+    return followingStatus;
+  }
+
+  renderFollowToggle() {
+    const { users, selectedUsers } = this.state;
+    if (users.length === 0) {
+      return '';
```
Comment (vaidehijoshi) [inline app/javascript/onboarding/components/FollowUsers.jsx:114]: Do we actually have to `return` an empty string here, or would just a simple `return` suffice?

Comment (maestromac) [reply app/javascript/onboarding/components/FollowUsers.jsx:114]: It does work without it but I would get `consistent-return` lint 🤷

File: app/javascript/onboarding/components/FollowUsers.jsx (hunk 7 above)
Comment (maestromac) [reply app/javascript/onboarding/components/FollowUsers.jsx:105]: yeah, can't wait for that day.
